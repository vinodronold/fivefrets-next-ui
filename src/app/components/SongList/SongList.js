import React, { PureComponent } from 'react'
import { bool, arrayOf, object } from 'prop-types'
import Link from 'next/link'
import { play } from '../../constants/routes'
import Card from '../Card'
import Loader from '../Loader'
import { ErrorBox } from '../Box'
import { Caption } from '../Typography'

export default class SongList extends PureComponent {
  static propTypes = {
    isError: bool,
    songs: arrayOf(object)
  }

  render() {
    const { isError = false, songs } = this.props
    return isError ? (
      <ErrorBox>
        <Caption>Oops.. ERROR.. Sorry.. please try refreshing... </Caption>
      </ErrorBox>
    ) : songs && songs.length > 0 ? (
      songs.map(s => (
        <Link prefetch key={s.id} href={play.href(s.id)} as={play.as(s.id)}>
          <Card id={s.id} title={s.title} />
        </Link>
      ))
    ) : (
      <Loader />
    )
  }
}
