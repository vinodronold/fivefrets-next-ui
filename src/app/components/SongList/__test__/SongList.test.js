import React from 'react'
import renderer from 'react-test-renderer'
import SongList from '../SongList'

jest.mock('next/router')

describe('components/SongList', () => {
  const _songs = [
    {
      1: { title: 'title 1' },
      2: { title: 'title 2' }
    }
  ]
  it('should match snapshot when song is Empty', () => {
    const _song_list = renderer.create(<SongList songs = {[]} FetchHomePageSongs={jest.fn()} />)
    expect(_song_list).toMatchSnapshot()
  })
  it('should match snapshot with songs', () => {
    const _song_list = renderer.create(<SongList songs={_songs} />)
    expect(_song_list).toMatchSnapshot()
  })
  it('should match snapshot with error', () => {
    const _song_list = renderer.create(<SongList isError={true} songs={_songs} />)
    expect(_song_list).toMatchSnapshot()
  })
})
