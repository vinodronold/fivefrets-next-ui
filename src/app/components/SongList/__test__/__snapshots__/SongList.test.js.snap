// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SongList should match snapshot when song is Empty 1`] = `
<div
  className="jsx-3649670468 container"
>
  <div
    className="jsx-3649670468 loader"
  >
    <div
      className="jsx-791847780 jsx-2291006144 bar"
    />
    <div
      className="jsx-791847780 jsx-3500035231 bar"
    />
    <div
      className="jsx-791847780 jsx-1901401244 bar"
    />
    <div
      className="jsx-791847780 jsx-40336921 bar"
    />
    <div
      className="jsx-791847780 jsx-3370661658 bar"
    />
  </div>
  <div
    className="jsx-4247411757 jsx-3754337244"
    style={Object {}}
  >
    LOADING...
  </div>
</div>
`;

exports[`components/SongList should match snapshot with error 1`] = `
<div
  className="jsx-2583165224 "
>
  <div
    className="jsx-4247411757 jsx-3754337244"
    style={Object {}}
  >
    Oops.. ERROR.. Sorry.. please try refreshing... 
  </div>
</div>
`;

exports[`components/SongList should match snapshot with songs 1`] = `
Array [
  <div
    className="jsx-3275184174"
    onClick={[Function]}
  >
    <div
      className="jsx-4231049512"
    >
      <img
        alt="YT"
        className="jsx-4231049512"
        height="auto"
        src="https://i.ytimg.com/vi/1/default.jpg"
        width="auto"
      />
    </div>
    <div
      className="jsx-250553575"
    >
      <div
        className="jsx-4247411757 jsx-873826105"
        style={Object {}}
      >
        title 1
      </div>
    </div>
  </div>,
  <div
    className="jsx-3275184174"
    onClick={[Function]}
  >
    <div
      className="jsx-4231049512"
    >
      <img
        alt="YT"
        className="jsx-4231049512"
        height="auto"
        src="https://i.ytimg.com/vi/2/default.jpg"
        width="auto"
      />
    </div>
    <div
      className="jsx-250553575"
    >
      <div
        className="jsx-4247411757 jsx-873826105"
        style={Object {}}
      >
        title 2
      </div>
    </div>
  </div>,
]
`;
